utils/core/src/serde/byte_reader.rs

```diff
@@ -92,6 +92,36 @@ pub trait ByteReader {
         Ok(u64::from_le_bytes(bytes))
     }
 
+    /// Returns a usize value read from `self` in ULEB128 format.
+    ///
+    /// # Errors
+    /// Returns a [DeserializationError] if a usize value could not be read from `self`.
+    fn read_usize(&mut self) -> Result<usize, DeserializationError> {
+        let mut result = u64::default();
+        let mut offset = 0;
+        loop {
+            let octet = self.read_u8()?;
+            // get the value from the octet and accumulate it in the result
+            result += (octet & OCTET_BIT_MASK) as u64 * 128u64.pow(offset);
+            // if the continuation bit equals 0, finish reading
+            if octet >> OCTET_VALUE_SIZE == 0 {
+                break;
+            }
+            offset += 1;
+        }
```

Comment:
If someone encodes a malicious value (where continuation bit is always one for many many bits) - how would this work? Should we maybe always break after 9 iterations?